crates/usvg/src/text/flatten.rs

```diff
@@ -19,58 +24,134 @@ fn resolve_rendering_mode(text: &Text) -> ShapeRendering {
     }
 }
 
+fn push_outline_paths(
+    span: &Span,
+    builder: &mut tiny_skia_path::PathBuilder,
+    new_children: &mut Vec<Node>,
+    rendering_mode: ShapeRendering,
+) {
+    let builder = mem::replace(builder, tiny_skia_path::PathBuilder::new());
+
+    if let Some(path) = builder.finish().and_then(|p| {
+        Path::new(
+            String::new(),
+            span.visibility,
+            span.fill.clone(),
+            span.stroke.clone(),
+            span.paint_order,
+            rendering_mode,
+            Arc::new(p),
+            Transform::default(),
+        )
+    }) {
+        new_children.push(Node::Path(Box::new(path)));
+    }
+}
+
 pub(crate) fn flatten(text: &mut Text, fontdb: &fontdb::Database) -> Option<(Group, NonZeroRect)> {
-    let mut new_paths = vec![];
+    let mut new_children = vec![];
 
-    let mut stroke_bbox = BBox::default();
     let rendering_mode = resolve_rendering_mode(text);
 
     for span in &text.layouted {
         if let Some(path) = span.overline.as_ref() {
-            stroke_bbox = stroke_bbox.expand(path.data.bounds());
             let mut path = path.clone();
             path.rendering_mode = rendering_mode;
-            new_paths.push(path);
+            new_children.push(Node::Path(Box::new(path)));
         }
 
         if let Some(path) = span.underline.as_ref() {
-            stroke_bbox = stroke_bbox.expand(path.data.bounds());
             let mut path = path.clone();
             path.rendering_mode = rendering_mode;
-            new_paths.push(path);
+            new_children.push(Node::Path(Box::new(path)));
         }
 
+        // Instead of always processing each glyph separately, we always collect
+        // as many outline glyphs as possible by pushing them into the span_builder
+        // and only if we encounter a different glyph, or we reach the very end of the
+        // span to we push the actual outline paths into new_children. This way, we don't need
+        // to create a new path for every glyph if we have many consecutive glyphs
+        // with just outlines (which is the most common case).
         let mut span_builder = tiny_skia_path::PathBuilder::new();
 
         for glyph in &span.positioned_glyphs {
-            if let Some(outline) = fontdb.outline(glyph.font, glyph.glyph_id) {
-                if let Some(outline) = outline.transform(glyph.transform) {
-                    span_builder.push_path(&outline);
+            // A COLRv0 glyph. Will return a vector of paths that make up the glyph description.
+            // TODO: Don't use black for foreground color? But not sure whether to use fill or stroke
```

Comment:
I don't think this affects SVG much, but I do think we should prefer a transparent background here.

Reply:
IIRC I can only choose the color but not the transparency... But I will check again when I'm on my computer.

Reply:
You're passing RGBA color here, so 0,0,0,255 should work just fine.

Reply:
Sorry, but I'm still not following... I'm passing `Color::black()`, which is defined as:

```rs
    /// Constructs a new `Color` set to black.
    #[inline]
    pub fn black() -> Color {
        Color::new_rgb(0, 0, 0)
    }
```

And then I'm doing:

```rs
fn paint_foreground(&mut self) {
        self.paint_color(ttf_parser::RgbaColor::new(
            self.foreground.red,
            self.foreground.green,
            self.foreground.blue,
            255,
        ));
    }
```

So isn't that already 0,0,0,255?